server/leafnode.go

```diff
@@ -813,8 +812,12 @@ func (s *Server) createLeafNode(conn net.Conn, rURL *url.URL, remote *leafNodeCf
 		var err error
 		acc, err = s.LookupAccount(lacc)
 		if err != nil {
-			s.Errorf("No local account %q for leafnode: %v", lacc, err)
+			// An account not existing is something that can happen with nats/http account resolver and the account
+			// has not yet been pushed, or the request failed for other reasons.
+			// remote needs to be set or retry won't happen
+			c.leaf.remote = remote
 			c.closeConnection(MissingAccount)
+			s.Errorf("No local account %s found for solicited leaf node connection", lacc)
```

Comment:
Looks like my comment about the error disappeared? I commented this:
It is likely that the account lookup failure means that the account does not exist, but there may be value in reporting the actual error. So error could be: `s.Errorf("Unable to lookup account %s for solicited leafnode connection: %v", lacc, err)`.